src/CompanyService.Broker/Consumers/GetUsersDepartmentsUsersPositionsConsumer.cs

```diff
@@ -0,0 +1,110 @@
+﻿using LT.DigitalOffice.CompanyService.Data.Interfaces;
+using LT.DigitalOffice.CompanyService.Models.Db;
+using LT.DigitalOffice.Kernel.Broker;
+using LT.DigitalOffice.Models.Broker.Models;
+using LT.DigitalOffice.Models.Broker.Requests.Company;
+using LT.DigitalOffice.Models.Broker.Responses.Company;
+using MassTransit;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.CompanyService.Broker.Consumers
+{
+    public class GetUsersDepartmentsUsersPositionsConsumer : IConsumer<IGetUsersDepartmentsUsersPositionsRequest>
+    {
+        private readonly IDepartmentUserRepository _departmentUserRepository;
+        private readonly IPositionUserRepository _positionUserRepository;
+
+        private List<UsersDepartment> GetDepartments(List<Guid> userIds)
+        {
+            List<UsersDepartment> response = new ();
+
+            List<DbDepartmentUser> dbDepartmentUsers = _departmentUserRepository.Find(userIds);
+
+            foreach (DbDepartmentUser dbDepartmentUser in dbDepartmentUsers)
+            {
+                UsersDepartment isInResponse = response.FirstOrDefault(x => x.Id == dbDepartmentUser.DepartmentId);
+
+                if (isInResponse != null)
+                {
+                    isInResponse.UserIds.Add(dbDepartmentUser.UserId);
+                }
+                else
+                {
+                    response.Add(
+                        new UsersDepartment(
+                            dbDepartmentUser.DepartmentId,
+                            dbDepartmentUser.Department.Name,
+                            new List<Guid>() { dbDepartmentUser.UserId }));
+                }
+            }
+            return response;
+        }
+
+        private List<UsersPosition> GetPositions(List<Guid> userIds)
+        {
+            List<UsersPosition> response = new();
+
+            List<DbPositionUser> dbPositionsUsers = _positionUserRepository.Find(userIds);
+
+            foreach (DbPositionUser dbPositionUser in dbPositionsUsers)
+            {
+                UsersPosition IsInResponse = response.FirstOrDefault(x => x.Id == dbPositionUser.PositionId);
```

Comment:
what are you doing here?

Reply:
I am checking if this UsersPosition is in my response list. If the position ID already exists in my response- add the userId to this UserPosition. Otherwise, I add a new UserPosition to my answer list